nexus/db-queries/src/db/datastore/switch_port.rs

```diff
@@ -312,319 +316,2021 @@ impl DataStore {
-                        enforce_first_as: p.enforce_first_as,
-                        vlan_id: p.vlan_id,
-                        communities: communities.into_iter().map(|c| c.community.0).collect(),
-                        allowed_import,
-                        allowed_export,
-                    };
+        self.transaction_retry_wrapper(
+            "switch_port_configuration_bgp_peer_community_remove",
+        )
+            .transaction(&conn, |conn| {
+                let parent_configuration = configuration.clone();
+                let settings_to_remove = community.clone();
+                let err = err.clone();
 
-                    result.bgp_peers.push(view);
-                }
+                async move {
+                    // resolve id of port_settings record
+                    let port_settings_id = switch_port_configuration_id(&conn, parent_configuration.clone())
+                        .await
+                        .map_err(|e: diesel::result::Error| {
+                            match e {
+                                diesel::result::Error::NotFound => {
+                                    err.bail(
+                                        Error::non_resourcetype_not_found(
+                                            format!("unable to lookup configuration with identifier {parent_configuration}")
+                                        )
+                                    )
+                                },
+                                _ => {
+                                    err.bail(Error::internal_error(
+                                        "error while looking up configuration for interface bgp peer"
+                                    ))
+                                },
+                            }
+                        })?;
 
-                // get the address configs
-                use db::schema::switch_port_settings_address_config::{
-                    self as address_config, dsl as address_config_dsl,
-                };
+                    // delete communities
+                    // PRIMARY KEY (port_settings_id, interface_name, addr, community)
+                    use db::schema::switch_port_settings_bgp_peer_config_communities as peer_communities;
+                    diesel::delete(peer_communities::table)
+                        .filter(peer_communities::port_settings_id.eq(port_settings_id))
+                        .filter(peer_communities::interface_name.eq(settings_to_remove.interface.to_string()))
+                        .filter(peer_communities::addr.eq(IpNetwork::from(bgp_peer)))
+                        .filter(peer_communities::community.eq(SqlU32::from(settings_to_remove.community)))
+                        .execute_async(&conn)
+                        .await
+                        .map_err(|e| {
+                            let message = "error while deleting community entry for bgp peer";
```

Comment:
When do you choose to use "deleting" vs "removing" (and "delete" vs "remove")?

See line 2329.

Reply:
More or less, `deleting` is referring to the action that are executing in the database (we are actually deleting rows), while the overall method / function is called "remove" because it is called by the `bgp-peer/remove` API.